src/AutoMoq/Unity/AutoMockingBuilderStrategy.cs

```diff
@@ -31,23 +31,18 @@ public override void PreBuildUp(IBuilderContext context)
         private void LoadAnyAbstractDependenciesOf(Type type)
         {
             foreach (var dependency in AbstractDependenciesOf(type))
-            {
-                var mock = CreateAMockTrackedByAutoMoq(dependency);
-                if (ThisTypeHasBeenRegisteredInIoC(dependency)) continue;
-                ioc.RegisterInstance(mock.ActualObject, dependency);
-            }
+                BuildThisByAskingTheContainerForIt(dependency);

```

Comment:
Fixing #29 caused failing tests around abstract dependencies.  Looking deeper into it, I realized that checking the container for the object was having a separate effect -- it was causing the object to be built by the AutoMoq processes injected into every type resolution.

That effect effectively does what I was trying to do here -- build the mock object and register it.  So went with that instead of manually building these objects.  I wrote more tests to cover this behavior.